simpeg_drivers/electromagnetics/time_domain_1d/driver.py

```diff
@@ -0,0 +1,132 @@
+# '''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
+#  Copyright (c) 2025 Mira Geoscience Ltd.                                          '
+#                                                                                   '
+#  This file is part of simpeg-drivers package.                                     '
+#                                                                                   '
+#  simpeg-drivers is distributed under the terms and conditions of the MIT License  '
+#  (see LICENSE file at the root of this source code package).                      '
+#                                                                                   '
+# '''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
+
+
+from __future__ import annotations
+
+import numpy as np
+from discretize import TensorMesh
+from discretize.utils import mesh_utils
+from geoh5py import Workspace
+from geoh5py.shared.merging.drape_model import DrapeModelMerger
+from geoh5py.ui_json.ui_json import fetch_active_workspace
+
+from simpeg_drivers.components.factories import MisfitFactory
+from simpeg_drivers.components.meshes import InversionMesh
+from simpeg_drivers.driver import InversionDriver
+from simpeg_drivers.utils.utils import topo_drape_elevation, xyz_2_drape_model
+
+from .params import (
+    TDEM1DForwardOptions,
+    TDEM1DInversionOptions,
+)
+
+
+class Base1DDriver(InversionDriver):
+    """Base 1D driver for electromagnetic simulations."""
+
+    _params_class = None
+    _validations = None
+
+    def __init__(self, workspace: Workspace, **kwargs):
+        super().__init__(workspace, **kwargs)
+
+        self.layers_mesh: TensorMesh = self.get_1d_mesh()
+        self.topo_z_drape = topo_drape_elevation(
+            self.params.data_object.vertices,
+            self.inversion_topography.locations,
+        )
+
+    @property
+    def inversion_mesh(self) -> InversionMesh:
+        """Inversion mesh"""
+        if getattr(self, "_inversion_mesh", None) is None:
+            # temp_workspace = Workspace()
+            with fetch_active_workspace(self.workspace, mode="r+"):
+                drape_models = []
+                temp_work = Workspace()
+                for part in self.params.data_object.unique_parts:
+                    indices = self.params.data_object.parts == part
+                    drape_models.append(
+                        xyz_2_drape_model(
+                            temp_work,
+                            self.topo_z_drape[indices],
+                            self.layers_mesh.h[0][::-1],
+                        )
+                    )
+
+                entity = DrapeModelMerger.create_object(self.workspace, drape_models)
+
+            self._inversion_mesh = InversionMesh(
+                self.workspace, self.params, entity=entity
+            )
+            self._inversion_mesh.layers_mesh = self.layers_mesh
```

Comment:
why are you setting this here?

Reply:
This is the 1D mesh and I need to be able to pass that information to `InversionData.simulation`, and currently InversionData and MisfitFactory don't have direct connection to the driver, so I need to attach it to something that gets through them all.